src/api/middleware/company.js

```diff
@@ -22,6 +24,32 @@ const verifyMaxConcurrentOffers = (owner, publishDate, publishEndDate) => async
     return next();
 };
 
+const verifyMaxConcurrentOffersOnCreate = (req, res, next) => {
+
+    if (req.body?.isHidden) return next();
+
+    const owner = req.user?.company || req.body.owner;
+    const publishDate = req.body.publishDate;
+    const publishEndDate = req.body.publishEndDate;
+
+    return verifyMaxConcurrentOffers(owner, publishDate, publishEndDate)(req, res, next);
+
+};
+
+const verifyMaxConcurrentOffersOnEdit = async (req, res, next) => {
+
+    if (req.body?.isHidden) return next();
+
+    const offer = await Offer.findById(req.params.offerId);
```

Comment:
Wrap this in a try catch, since the call to the db can fail. Additionally, you should use the service and not direct db calls
```suggestion
    try {
                const offer = await (new OfferService()).getOfferById(req.params.offerId, req.user);
                if (!offer) throw new APIError(HTTPStatus.NOT_FOUND, ErrorTypes.VALIDATION_ERROR, ValidationReasons.OFFER_NOT_FOUND(req.params.offerId));
            } catch (err) {
                console.error(err);
                return next(err);
            }
```

Reply:
By doing this, you can avoid the ?. operators later, since inside the try block, the offer always exists.